ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -832,6 +873,29 @@ public static void accountAndContainerNamePreconditionCheck(RestRequest restRequ
     }
   }
 
+  /**
+   * Validate the signed path in the URI if required for specific {@link Container}.
+   * @param restRequest the {@link RestRequest} that may contain signed path.
+   * @param expectSignedPath the expected signed path specified in {@link com.github.ambry.config.FrontendConfig}.
+   * @throws RestServiceException
+   */
+  public static void validateSignedPathIfRequired(RestRequest restRequest, String expectSignedPath)
+      throws RestServiceException {
+    Container targetContainer = getContainerFromArgs(restRequest.getArgs());
+    String signedPathInUri = getSignedPath(restRequest);
+    if (targetContainer.isSignedPathValidationRequired()) {
+      if (!signedPathInUri.equals(expectSignedPath)) {
+        throw new RestServiceException("Signed path in restRequest doesn't match the expected one",
+            RestServiceErrorCode.AccessDenied);
+      }
+    } else {
+      String signedIdPrefixWithNoSlash = SIGNED_ID_PREFIX.substring(0, SIGNED_ID_PREFIX.indexOf("/"));
+      if (!signedPathInUri.isEmpty() && !signedPathInUri.equals(signedIdPrefixWithNoSlash)) {
+        throw new RestServiceException("Invalid path is found in URI", RestServiceErrorCode.BadRequest);
```

Comment:
This `if condition` checks invalid path in URI when container doesn't require signed path validation and it is not a signed id.  Thus, `/orNotToRemove/blobId` will be rejected here.